Server/serverSlate.go

```diff
@@ -36,21 +59,49 @@ func main() {
 	fmt.Println("Got connection from:")
 	fmt.Println(conn.LocalAddr().String())
 
-	hostReader := bufio.NewReader(conn)
-	timeStamps := make([]string, KEY_PRESS_COUNT+1)
+	timestamps := make([]int64, KEY_PRESS_COUNT+1)
+
+	hostCommunicator := json.NewDecoder(conn)
 
-	message, _ := hostReader.ReadString('*') // using * as delim and hardcoded in app to append * at end of message
-	fmt.Println(message)
-	if message == "started capture"+"*" {
-		simulateKeyPress(KEY, KEY_PRESS_COUNT, timeStamps)
+	var hostAction action
+
+	for {
+		err := hostCommunicator.Decode(&hostAction)
+		fmt.Println(err)
+		if hostAction.Code == 1 {
```

Comment:
define constants for these command codes, eg ACTION_CAPTURE_STARTED and ACTION_CAPTURE_RESULTS